### PR TITLE
Fixing https://wso2.org/jira/browse/APIMANAGER-4482 : REST API Store subscription issue

### PR DESCRIPTION
**Pull request by malinthaprasan**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java [hunk 1]
```diff
@@ -2849,28 +2849,19 @@ public static NativeObject jsFunction_getSubscriber(Context cx, Scriptable thisO
     }
 
     public static boolean jsFunction_addSubscriber(Context cx, Scriptable thisObj, Object[] args, Function funObj)
-            throws ScriptException, APIManagementException, UserStoreException {
+            throws ScriptException, APIManagementException{
 
         if (args != null && isStringArray(args)) {
-            Subscriber subscriber = new Subscriber((String) args[0]);
+            String username = (String) args[0];
             String groupId = (String) args[1];
```
Comment (shaniR) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: would be better to check args length

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java [hunk 2]
```diff
@@ -2849,28 +2849,19 @@ public static NativeObject jsFunction_getSubscriber(Context cx, Scriptable thisO
     }
 
     public static boolean jsFunction_addSubscriber(Context cx, Scriptable thisObj, Object[] args, Function funObj)
-            throws ScriptException, APIManagementException, UserStoreException {
+            throws ScriptException, APIManagementException{
 
         if (args != null && isStringArray(args)) {
-            Subscriber subscriber = new Subscriber((String) args[0]);
+            String username = (String) args[0];
```
Comment (shaniR) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Would be better to check args length

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java [hunk 3]
```diff
@@ -936,6 +936,25 @@ public void addSubscriber(Subscriber subscriber, String groupingId)
         apiMgtDAO.addSubscriber(subscriber, groupingId);
     }
 
+    public void addSubscriber(String username, String groupingId)
```
Comment (msmshariq) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java]: Some of the public methods in this class doesn't have Javadoc comments.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/PostAuthenticationInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                    }
+                    return;
+                }
+                apiConsumer.addSubscriber(username, groupId);
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Subscriber " + username + " added to AM_SUBSCRIBER database");
+                }
+            }
+        } catch (APIManagementException e) {
+            String errorMsg = "Unable to add the subscriber " + username;
```
Comment (chrishantha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/PostAuthenticationInterceptor.java]: Put the error message directly in the method as the `errorMsg` is not used anywhere.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java (hunk 3 above)
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java]: Please note all the public method JavaDocs (which does not already have) are inherited from https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIManager.java class.

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java (hunk 2 above)
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Fixed in #1581 (https://github.com/wso2/carbon-apimgt/commit/eb1deba65015d972478713ccf5557bc57be73612)

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/PostAuthenticationInterceptor.java (hunk 4 above)
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/PostAuthenticationInterceptor.java]: Fixed in #1581 (https://github.com/wso2/carbon-apimgt/commit/26b9dc6a8281dcd9b7bad6ae7f8c53a7e7cf5854)

**Pull request merged**